katdal/dataset.py

```diff
@@ -190,6 +210,46 @@ def __hash__(self):
         """Base hash on description tuple, just like equality operator."""
         return hash(self._description)
 
+    def subset(self, first, last):
+        """Get a new :class:`SpectralWindow` representing a subset of the channels.
+
+        The returned :class:`SpectralWindow` covers the same frequencies as
+        channels [first, last) of the original.
+
+        Raises
+        ------
+        IndexError
+            If [first, last) is not a (non-empty) subinterval of the channels
+        """
+        if not (0 <= first < last <= self.num_chans):
+            raise IndexError('indices out of range')
```

Comment:
I would make this slightly more specific (e.g. `channel indices out of range`), since peeps tend not to look at the traceback but rather at the error message :-)

Reply:
Done.